app/models/user.rb

```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  devise :database_authenticatable, :registerable, :confirmable,
+         :recoverable, :rememberable, :validatable
```

Comment:
cơ chế hoạt động của module `rememberable` trong devise như thế nào em?
nó dùng trường nào để lưu giá trị? cách kiểm tra ntn?

Reply:
- rememberable dùng cookies để nhớ người dùng với thời gian ghi nhớ mặc định là 2 tuần.
Tạo khóa cookie bằng id, remmember_value (Cái này được tạo ra bằng cách băm 29 kí tự của password) và thời gian
- Nó lưu remember_created_at trong database
- Khi kiểm tra, so sáng giá trị thời gian rồi so sánh remember_value
        # The token is only valid if:
        # 1. we have a date
        # 2. the current time does not pass the expiry period
        # 3. the record has a remember_created_at date
        # 4. the token date is bigger than the remember_created_at
        # 5. the token matches